src/Xamarin.Android.Build.Tasks/Tasks/Generator.cs

```diff
@@ -181,7 +181,7 @@ protected override string GenerateCommandLineCommands ()
 				if (!string.IsNullOrEmpty (EnumMetadataDirectory))
 					WriteLine (sw, $"--enummetadata=\"{EnumMetadataDirectory}\"");
 				if (!string.IsNullOrEmpty (AssemblyName))
-					WriteLine (sw, $"--assembly={AssemblyName}");
+					WriteLine (sw, $"--assembly=\"{AssemblyName}\"");
```

Comment:
This could use `MonoAndroidHelper.QuoteFileNameArgument`, so that we have quoting code in a single place

Reply:
Its a good idea. However, this file is full off "\"" entires, so I feel we would need to change all of them. 
Also `QuoteFileNameArgument` creates a new `CommandLineBuilder` every time its called, it feels like just hard coding the quotes at this point would be more efficient? 🤷